stgit/utils.py

```diff
@@ -92,7 +94,31 @@ def hook(*parameters):
         argv.extend(parameters)
 
         # On Windows, run the hook using "bash" explicitly
-        if os.name != 'posix':
+        if os.name == 'nt':
+            # try to locate bash.exe on the user's PATH, but don't fall for the
+            # WSL shim/bootstrapper %SYSTEMROOT%/System32/bash.exe
+            path = os.environ['PATH']
+            try:
+                # remove %SYSTEMROOT%/System32 from PATH
+                sys32 = os.path.join(os.environ['SYSTEMROOT'], 'system32').lower()
+                os.environ['PATH'] = ';'.join(
+                    (p for p in path.split(';') if not p.lower() == sys32)
+                )
+                # try to honour the user's PATH (sans System32)
+                bash_exe = shutil.which('bash.exe')
+                if not bash_exe:
+                    # failed, try to use bash.exe that comes with Git for Windows
+                    git_exe = shutil.which('git.exe')
+                    if not git_exe:
+                        raise StgException(
+                            'Failed to locate either bash.exe or git.exe'
+                        )
+                    bash_exe = pathlib.Path(git_exe).parents[1] / 'bin' / 'bash.exe'
+                argv.insert(0, str(bash_exe))
+            finally:
+                # restore PATH
+                os.environ['PATH'] = path
```

Comment:
This all feels fine. It's ugly as hell but at least it's gated behind `if os.name == 'nt':` :joy: